contracts/AutomationBotAggregator.sol

```diff
@@ -0,0 +1,209 @@
+            AUTOMATION_AGGREGATOR_BOT_KEY
+        );
+
+        // TODO: replace decode with IValidator(validatorAddress).decode
+        address validatorAddress = getValidatorAddress(groupTypeId);
+        (uint256[] memory cdpIds, uint256[] memory triggerTypes) = decode(triggersData);
+
+        /*         require(
+            IValidator(validatorAddress).validate(replacedTriggerId, triggersData),
+            "aggregator/validation-error"
+        );
+        (uint256[] memory cdpIds, uint256[] memory triggerTypes) = IValidator(validatorAddress)
+            .decode(triggersData); */
+
+        uint256 firstTriggerId = AutomationBot(automationBot).triggersCounter() + 1;
+
+        uint256[] memory triggerIds = new uint256[](triggersData.length);
+        for (uint256 i = 0; i < triggerTypes.length; i += 1) {
+            (bool status, ) = automationBot.delegatecall(
+                abi.encodeWithSelector(
+                    AutomationBot(automationBot).addTrigger.selector,
+                    cdpIds[i],
+                    triggerTypes[i],
+                    replacedTriggerId[i],
+                    triggersData[i]
+                )
+            );
+
+            triggerIds[i] = firstTriggerId + i;
+            require(status, "aggregator/add-trigger-fail");
+        }
+
+        BotAggregatorLike(automationAggregatorBot).addRecord(cdpIds[0], groupTypeId, triggerIds);
+    }
+
+    function removeTriggerGroup(
+        uint256 cdpId,
+        uint256 groupId,
+        uint256[] memory triggerIds,
+        bool removeAllowance
+    ) external onlyDelegate {
+        address automationBot = serviceRegistry.getRegisteredService(AUTOMATION_BOT_KEY);
+        address automationAggregatorBot = serviceRegistry.getRegisteredService(
+            AUTOMATION_AGGREGATOR_BOT_KEY
+        );
+
+        for (uint256 i = 0; i < triggerIds.length; i += 1) {
+            (bool status, ) = automationBot.delegatecall(
+                abi.encodeWithSelector(
+                    AutomationBot(automationBot).removeTrigger.selector,
+                    cdpId,
+                    triggerIds[i],
+                    removeAllowance
+                )
+            );
+            require(status, "aggregator/remove-trigger-fail");
+        }
+        BotAggregatorLike(automationAggregatorBot).removeRecord(cdpId, groupId, triggerIds);
+    }
+
+    function addRecord(
+        uint256 cdpId,
+        uint16 groupTypeId,
+        uint256[] memory triggerIds
+    ) external {
+        ManagerLike manager = ManagerLike(serviceRegistry.getRegisteredService(CDP_MANAGER_KEY));
+
+        require(isCdpAllowed(cdpId, msg.sender, manager), "aggregator/no-permissions");
+
+        triggerGroupCounter = triggerGroupCounter + 1;
+        activeTriggerGroups[triggerGroupCounter] = TriggerGroupRecord(
+            getTriggerGroupHash(cdpId, triggerGroupCounter, triggerIds),
+            cdpId,
+            groupTypeId
```

Comment:
why do we need `groupTypeId`? can You add it in getTriggerGroupHash? that would save one storage slot.

Reply:
Or even ignore groupTypeId in removal all along